### ISSUE TITLE
python 3.13.0rc3 and collections.abc

### ISSUE THREAD

**Issue by stanislavlevin** [pylint-dev/pylint#10000]
### Bug description

False positive error of missing name in `collections.abc`.
For example linting over the code:

```
from collections.abc import MutableMapping
```
reports:
```
No name 'MutableMapping' in module 'collections.abc' (no-name-in-module)
```

This error started with updating `Python` from `3.13.0rc2` to `3.13.0rc3`.
Most likely caused by https://github.com/python/cpython/pull/124735

### Command used

pylint

### Pylint output

No name 'MutableMapping' in module 'collections.abc' (no-name-in-module)

### Expected behavior

No error

### Pylint version

Python 3.13.0rc3
astroid-3.3.4
pylint-3.3.1

**Comment by julian-smith-artifex-com** [comment]
I've just seen almost exactly the same problem today, with `from collections.abc import Sequence`.

**Comment by nitzmahone** [comment]
Yep, it looks like the static analysis can't follow the [runtime-swap-a-dee-doo](https://github.com/python/cpython/pull/124735/files#diff-8f5b4b1ece612d08455a0a88823f7816e4f0c56a90104721478ff07045f57a9dR3) that's now happening to the guts of `collections.abc`. That's a gnarly one...

**Comment by jacobtylerwalls** [comment]
Fixed in astroid 3.3.5 if anyone needs this right away.

**Comment by nitzmahone** [comment]
Yep, works for us now- thanks for the quick fix and release!

**Comment by cavo789** [comment]
Hello

I got the same (I think) i.e. I'm using f.i. Callable (from typing import Callable) and pylint ask to replace by collections.abc.Callable but that module seems to not exists.

I'm using a Docker base image Python 3.13.0.

Pylint seems to be up to date (i've fired pip install --upgrade), can you guide me please?

What should I do to be able to respect suggestions from pylint?

Thanks!

**Comment by Pierre-Sassoulas** [comment]
>  collections.abc.Callable but that module seems to not exists.

Not sure what the problem is but I think it exists since python 3.9 ; https://docs.python.org/3.9/library/collections.abc.html#collections.abc.Callable (also in 3.13 : https://docs.python.org/3.13/library/collections.abc.html#collections.abc.Callable)

**Comment by jacobtylerwalls** [comment]
What is your version of astroid? You would need 3.3.5 to get this fix. There has not yet been a patch release of pylint to enforce that.

**Comment by cavo789** [comment]
So fast, thanks ! 

Ok so perhaps my editor is incorrectly configured. 

Vscode throws that error while editing my .py script. I should admit that i've trust him (not fired the script in my console, will do tomorrow).

I got the error for Callable and for Sequence too...

If it helps, i'm using a  .pylintrc file and start pylint in my console.

Thanks for any suggestions...

**Comment by cavo789** [comment]
> What is your version of astroid? You would need 3.3.5 to get this fix. There has not yet been a patch release of pylint to enforce that.

Oh sorry i'm in a train (back to home) and was trying to find ressources so I can continue on this tomorrow... 

I'm not sure about the version of astroid (will check tomorrow) and more didn't know what astroid is (i'm a junior in python's world). 

I'm using python 3.13.0 (docker image) and i've installed pylint days ago with pip install pylint.

**Comment by Pierre-Sassoulas** [comment]
Try doing something like ``pip install astroid -U`` on pip (i.e. upgrade astroid to 3.3.5 explicitly)

**Comment by cavo789** [comment]
Hi all

Here is my current configuration:

```bash
> /usr/local/bin/python3 --version

Python 3.13.0
```

```bash
> pylint --version

pylint 3.3.1
astroid 3.3.5
Python 3.13.0 (main, Nov 12 2024, 02:27:08) [GCC 12.2.0]
```

(just to be sure, I've well fired `pip install astroid -U`; no change)

By running pylint on the command line here is some errors I got:

```text
************* Module core.Entity
src/core/Entity.py:238:30: W6001: 'typing.Callable' is deprecated, use 'collections.abc.Callable' instead (deprecated-typing-alias)
src/core/Entity.py:303:33: W6001: 'typing.Callable' is deprecated, use 'collections.abc.Callable' instead (deprecated-typing-alias)
```

Then, I edit that file and replace `from typing import Callable` .

My trouble comes from VSCode:

![Image](https://github.com/user-attachments/assets/eb1e08a7-0dbb-42e4-a6a3-1ba19f74f91e)

I'm using the latest version of Pylint addon i.e. v2024.1.13171016 (last updated: 2024-11-13, 08:01:24).

Perhaps a setting in VSCode; I'll search in that direction.

**Comment by cavo789** [comment]
Solved. It was well a setting to add in my vscode `settings.json` file : `"pylint.args": [ "--rcfile /app/.config/.pylintrc"]` i.e. the path to my configuration file and where I've set the minimum version of Python with that line `py-version = 3.13.0`.

Thanks for the support guys!

**Comment by antoni-szych-rtbhouse** [comment]
Seems like it stopped working in 3.13.1 again :(

public repo jobs:
python 3.13.0, pylint (3.3.2), astroid (3.3.5): https://github.com/rtbhouse-apps/rtbhouse-python-sdk/actions/runs/12123430319/job/33798902493
python 3.13.1, pylint (3.3.2), astroid (3.3.5): https://github.com/rtbhouse-apps/rtbhouse-python-sdk/actions/runs/12171673933/job/33949004704

```
rtbhouse_sdk/client.py:6:0: E0401: Unable to import 'collections.abc' (import-error)
```

**Comment by hofrob** [comment]
Can confirm that something changed in 3.13.1. 

- python 3.13.1
- astroid==3.3.6

```python
import astroid
ast = astroid.MANAGER.ast_from_string("from collections import abc")
import_node = list(ast.get_children())[0]
import_node.do_import_module("collections").import_module("abc", relative_only=True)
```

I'm doing some custom linting and now I have to set `relative_only=False` only for the `collections` package. Since I don't fully understand what this flag does, I might just be doing this incorrectly though 😬

**Comment by jacobtylerwalls** [comment]
Yep, we're tracking the new problem in #10112
